pyproject.toml

```diff
@@ -32,11 +32,13 @@ black = "^23.12.1"
 colorama = "^0.4.6"
 pinecone-client = {version="^3.0.0", optional = true}
 regex = "^2023.12.25"
+spacy = "^3.0"
```

Comment:
Given `spacy` is a heavier library, could we make it an optional dependency within the `processing` extras?